simplq/src/store/selectedQueue/selectedQueueSlice.js

```diff
@@ -10,8 +10,14 @@ const selectedQueueSlice = createSlice({
     queueName: null,
     queueCreationTimestamp: null,
     tokens: [],
+    queueDetails: {},
+  },
+  reducers: {
+    updateQueueStatus: (state, action) => {
+      state.queueDetails.isActive = action.isActive;
+      state.queueDetails.description = action.description;
+    },
   },
```

Comment:
I think you should define this in the extraReducers section as 
```
 [pauseQueue.fulfilled]: (state, action) => {
      state.queueDetails.isActive = action.isActive;
      state.queueDetails.description = action.description;
      return state;
 },
```

That's the pattern we use everywhere else, and we only want to set these values if the action is successful after all. You can take a look at [this](https://github.com/SimplQ/simplQ-frontend/blob/master/simplq/src/store/index.js), and see where the reducers are imported from. You'll notice that they are all defined specific to the status of the action. Keep in mind that the name of the thunk you defined in ayncActions should be used in extraReducer (currently pauseQueue).

Reply:
Sorry for all the questions, still learning this type of reducer implementation. I updated with your suggestions here, now how do I use the `queueDetails` fields in the `PauseQueue` component now?  Or should I be using my local variables there and then set the `queueDetails` with those?

Reply:
No worries, it took me a long time to figure this stuff out. I just noticed that the response of the setQueueStatus API call contains just queueID, queueName and status. It doesn't contain isActive or description. So the action parameter in the extraReducer above doesn't contain isActive or description, it'll just return undefined all the time. I suggest storing just status in the selectedQueueSlice and creating the isActive and description variables conditionally using a selector called selectQueueStatus that returns this status. isActive and description need not be local state variables since when the status from the selector changes, a re-render is triggerred, so isActive and description should change as well.

```
  initialState: {
   ....
   queueStatus: ''
  },
  ....,
 [pauseQueue.fulfilled]: (state, action) => {
      state.queueDetails.status = action.status;
      return state;
 },
```
Is the structure clearer now?

Reply:
give me some time to think about this

Reply:
@maaverik you are right. I'll suggest some more changes on top of what you've said.

1. Set initialState on line 8 to `{}`
2. Change reducer which handles fulfilled request to save the entire backend response into the state:
```
    // handle fulfiled request
    [getSelectedQueue.fulfilled]: (state, action) => {
      return action.payload;
    },
```
3. Remove the `[getSelectedQueue.rejected]` reducer, we don't need it.
4. On updateQueueStatus.fulfilled, update just the queue status:
```
 [updateQueueStatus.fulfilled]: (state, action) => {
      state.status = action.payload.status;    <---- don't forget the .payload.
      return state;
 },
```

Reply:
It would look like this at the end:

```
extraReducers: {
    // handle fulfiled request
    [getSelectedQueue.fulfilled]: (state, action) => {
      return action.payload;
    },
    // add newly created token to currently selected queue's token list
    [joinQueue.fulfilled]: (state, action) => {
      state.tokens.push(action.payload);
      return state;
    },
    // remove deleted token from currently selected queue's token list
    [deleteToken.fulfilled]: (state, action) => {
      state.tokens = state.tokens.filter((token) => token.tokenId !== action.payload.tokenId);
      return state;
    },
    // update queue status on updates
    [updateQueueStatus.fulfilled]: (state, action) => {
      state.status = action.payload.status;
      return state;
    },
  },
```

Note: Backend was not sending the status when you get the queue, I've fixed it [here](https://github.com/SimplQ/simplQ-backend/pull/125) just now, so it should work now.